### PR TITLE
feat(cards): Add support for cards & card orders

### PR DESCRIPTION
**Pull request by jlincodes**
## What
Adds support for cards and card orders

## Why
Allows customers to interact with the Cards API via the Ruby SDK

### CODE REVIEW

File: lib/lob/resources/card.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+module Lob
+  module Resources
+    class Card < Lob::Resources::ResourceBase
+
+      def initialize(config)
+        super(config)
+        @endpoint = "cards"
+      end
+      
+      def update(resource_id, body={}, headers={})
```
Comment (julielinter) [inline lib/lob/resources/card.rb:12]: I noticed that update isn't available for the other resources (which is consistent with the Node SDK as well) so I only added it here. I'm not sure what the reasoning for leaving out update functionality is. However, we added `/cards` update for Node, PHP and Python SDK, so we might as well add it here and be consistent.

File: examples/README.md [hunk 2]
```diff
@@ -38,6 +38,14 @@ Please note that if you are running this with a Test API Key, the verification A
 bundle exec ruby examples/csv_verify/verify.rb
 ```
 
+## /cards.rb
+
+An example showing how to create a check using Lob's [Cards API](https://docs.lob.com/#tag/Cards).
```
Comment (ross-martin) [inline examples/README.md:43]: 🍝 `check`

File: lib/lob/resources/card.rb (hunk 1 above)
Comment (ajorczak-lob) [reply lib/lob/resources/card.rb:12]: As long as the endpoint is publically exposable, it should be added.

File: examples/cards.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
+$:.unshift File.expand_path("../lib", File.dirname(__FILE__))
+require 'lob.rb'
+require 'pp'
+
+# initialize Lob object
+lob = Lob::Client.new(api_key: 'YOUR_API_KEY')
```
Comment (ajorczak-lob) [inline examples/cards.rb:6]: Let's update this just for some best practice reasons:
```suggestion
LOB_LIVE_API_KEY = ENV['LOB_LIVE_API_KEY']
lob = Lob::Client.new(api_key: LOB_LIVE_API_KEY)
```

**Review by ajorczak-lob — CHANGES_REQUESTED** [review]
Let's update the `example/cards.rb` file for some better practices.

File: lib/lob/resources/card.rb (hunk 1 above)
Comment (julielinter) [reply lib/lob/resources/card.rb:12]: For cards, update is publicly available. And I saw after creating this PR that there are endpoints where update is not publicly available, so I answered my own question (who would have thunk that reading docs would answer one's API questions 🙃 )

**Comment by mwj8410** [comment]
This PR has gone stale, so I've pushed a couple of updates to it as follows:

- Test cases were expanded to exercise a couple of additional functions available to this resource
- A few formatting differences between the test file and other files were addressed
- Uncommented examples and added additional examples to show all operations available
- Added file linking for Cards that has been put in place for other resources in `master`
- Resolved conflicts with master

@ajorczak-lob has your request for better practices in the example been addressed? I'm unsure what precisely you had in mind

@bethqiang do you have any concern with the PR being merged?

**Review by ajorczak-lob — APPROVED** [review]
LGTM. Thanks for taking the time to update!

**Comment by ajorczak-lob** [comment]
> @ajorczak-lob has your request for better practices in the example been addressed? I'm unsure what precisely you had in mind

Yes, I think so. Thanks for cleaning this up.

**Comment by julielinter** [comment]
> This PR has gone stale, so I've pushed a couple of updates to it as follows:

Thank you for updating this @mwj8410 🙏🏻  I've been meaning to get back to this but couldn't find the time to do so 😭

**Pull request merged**